### PR TITLE
Register DocBlockDriver

### PR DESCRIPTION
**Pull request by simPod**
Registers driver introduced in https://github.com/schmittjoh/serializer/pull/1214

### CODE REVIEW

File: DependencyInjection/JMSSerializerExtension.php [hunk 1]
```diff
@@ -108,6 +109,13 @@ public function loadInternal(array $config, ContainerBuilder $container)
             $container->setParameter('jms_serializer.infer_types_from_doctrine_metadata', false);
         }
 
+        if (class_exists(DocBlockDriver::class)) {
```
Comment (simPod) [inline DependencyInjection/JMSSerializerExtension.php:112]: maybe this should be registered after typed properties driver?

**Comment by goetas** [comment]
rebased and merged with https://github.com/schmittjoh/JMSSerializerBundle/commit/5adc373ad1d689bae6b65c5a3dec6ec35df74e6a

**Comment by alexander-schranz** [comment]
This seems currently crash exist project which are not supporting DocBlocks be used by the jms serializer. I would appreciate making the DocBlockDriver optional. #845

**Comment by simPod** [comment]
@alexander-schranz what is the problem?

**Comment by alexander-schranz** [comment]
@simPod it tries to read PHPDocs which was not be written to be used by the serializer. See #844

**Comment by simPod** [comment]
Unfortunately, there's no code snippet or test. TBH I'd rather check whether this could be fixed at the source rather than adding more complexity to configuration first. Without failing test it's difficult.

**Comment by alexander-schranz** [comment]
Its just that the DocBlockDriver fails hardly when it can not read anything. But in our case the DocBlockDriver never should be used to detect anything from the DocBlocks it should behave the same like before.

So if you have example:

```php
class SomeVendor {
     /**
      * @var SomeFalseType
      */
     private $property;
}
```

And serialize this:

```php
$this->serializer->serialize($myModel);
```

It fails as the `DocBlockDriver` reads the data  `@var SomeFalseType`.

I understand your point but I would more go with currently add the option to make it activateable to avoid this bc break and deprecate it when its disabled and in the next major release the `DocBlockDriver` can be used by default and the option can be removed again.

**Comment by simPod** [comment]
Ok now it makes more sense. It's up to @schmittjoh now I guess. 

Though if there's such issue in DocBlockDriver maybe you should open issue in serializer lib as well.

**Comment by alexander-schranz** [comment]
I still think this should be opt-in. Because maybe there is a valid type defined, but actually it is a false type and when you update from 3.8 to 3.9 the type then is changing. So I actually think from Backward Compatibility view the DocBlockDriver should be not activated by default as this as we see can have negative side effects. 

In our case it also appears for mixed union type e.g.:

```php
    /**
     * @var string|bool|array
     * @Groups({"frontend"})
     */
    private $value;
```

Ends with:

> Can't use union type string|bool|array for collection in Sulu\Component\Content\Compat\PropertyParameter:value

**Comment by lchrusciel** [comment]
Can confirm existence of similar issue in Sylius with the following error:
image

**Pull request closed without merging**